### DISCUSSION TITLE
Fish shell with python virtual environements

### DISCUSSION THREAD

**Discussion by andreichalapco** [fish-shell/fish-shell#9270] [Q&A] [answered]
Hi.

I've stumbled across the following problem. How can I configure my fish shell to show me in parantheses (like in the bash shell) the python virtual environment I'm currently using ... ?

**Comment by faho** [comment, marked as answer]
Python's [activate.fish script](https://github.com/python/cpython/blob/main/Lib/venv/scripts/posix/activate.fish) does this by default, unless you have set $VIRTUAL_ENV_DISABLE_PROMPT.

So either you are using some other method to activate your virtual environment, or the prompt you are using disables it, presumably to show it elsewhere. E.g. our "disco" sample disables it to display the venv itself on the right side:

https://github.com/fish-shell/fish-shell/blob/70d2e97c9ac566893b159963131a1d0c80000da7/share/tools/web_config/sample_prompts/disco.fish#L84-L90

**Reply by andreichalapco** [reply]
Thx for the reply. I use direnv to load my python virtual environment ...

**Reply by faho** [reply]
In that case, I'd see what direnv does to do that. I'd assume it doesn't run the activate.fish script in the current shell, so you would have to add the prompt part yourself.

See if $VIRTUAL_ENV is set to the correct value and then add it to your fish_prompt. Something like

```fish
if test -n "$VIRTUAL_ENV"
    echo -n (set_color blue)"($VIRTUAL_ENV)"(set_color normal)
end
```
